src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -731,6 +734,33 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     }
   }
 
+
+  private def getClusterNetwork(projectLabels: Map[String, String]): Option[VPCNetworkName] = {
+    //if we specify a network label in config, we should use that to see if the project has a network specified
+    //if it does, we will use that, else we will look to see if leo config has a network specified, else we will use the default
+    dataprocConfig.projectVPCNetworkLabel match {
```

Comment:
```
dataprocConfig.projectVPCNetworkLabel.flatMap(projectLabels.get).orElse(dataprocConfig.vpcNetwork).map(VPCNetworkName)
```
?

Reply:
my brain is really breaking on reading "project labels dot get or else" without it actually being `.getOrElse`, and i'm low-level anxious about which way to associate the three function calls. i'll take a poke at this and see if i can come up with something that inspires more chill in me